agglayer/proof_generation_error.go

```diff
@@ -0,0 +1,657 @@
+package agglayer
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+var errNotMap = errors.New("inner error is not a map")
+
+const (
+	InvalidSignerErrorType                = "InvalidSigner"
+	InvalidPreviousLERErrorType           = "InvalidPreviousLocalExitRoot"
+	InvalidPreviousBalanceRootErrorType   = "InvalidPreviousBalanceRoot"
+	InvalidPreviousNullifierRootErrorType = "InvalidPreviousNullifierRoot"
+	InvalidNewLocalExitRootErrorType      = "InvalidNewLocalExitRoot"
+	InvalidNewBalanceRootErrorType        = "InvalidNewBalanceRoot"
+	InvalidNewNullifierRootErrorType      = "InvalidNewNullifierRoot"
+	InvalidImportedExitsRootErrorType     = "InvalidImportedExitsRoot"
+	MismatchImportedExitsRootErrorType    = "MismatchImportedExitsRoot"
+	InvalidNullifierPathErrorType         = "InvalidNullifierPath"
+	InvalidBalancePathErrorType           = "InvalidBalancePath"
+	BalanceOverflowInBridgeExitErrorType  = "BalanceOverflowInBridgeExit"
+	BalanceUnderflowInBridgeExitErrorType = "BalanceUnderflowInBridgeExit"
+	CannotExitToSameNetworkErrorType      = "CannotExitToSameNetwork"
+	InvalidMessageOriginNetworkErrorType  = "InvalidMessageOriginNetwork"
+	InvalidL1TokenInfoErrorType           = "InvalidL1TokenInfo"
+	MissingTokenBalanceProofErrorType     = "MissingTokenBalanceProof"
+	DuplicateTokenBalanceProofErrorType   = "DuplicateTokenBalanceProof"
+	InvalidSignatureErrorType             = "InvalidSignature"
+	InvalidImportedBridgeExitErrorType    = "InvalidImportedBridgeExit"
+	UnknownErrorType                      = "UnknownError"
+)
+
+type PPError interface {
+	String() string
+}
+
+// ProofGenerationError is a struct that represents an error that occurs when generating a proof.
+type ProofGenerationError struct {
+	GenerationType string
+	InnerErrors    []PPError
```

Comment:
Do you even need the `PPError` interface?

```suggestion
	InnerErrors    []error
```

Reply:
I need an interface because of different errors (different structs). I need a way to unmarshal different errors, and that is why I need an interface here, since there are 50+ different errors.